src/Translator/TranslatableMessageInterface.php

```diff
@@ -0,0 +1,46 @@
+<?php
+/**
+ * Zend Framework (http://framework.zend.com/)
+ *
+ * @link      http://github.com/zendframework/zf2 for the canonical source repository
+ * @copyright Copyright (c) 2005-2015 Zend Technologies USA Inc. (http://www.zend.com)
+ * @license   http://framework.zend.com/license/new-bsd New BSD License
+ */
+
+namespace Zend\Validator\Translator;
+
+/**
+ * This interface provide methods for extract the translatable parts of a message.
+ */
+interface TranslatableMessageInterface
+{
+    /**
+     * Returns the message to translate.
+     *
+     * The message may include placeholders for message variable interpolation.
+     *
+     * @return string
+     */
+    public function getMessageTemplate();
+
+    /**
+     * Returns the variables to be replaced in the message template placeholders.
+     *
+     * @return array
+     */
+    public function getMessageVariables();
+
+    /**
+     * Returns the translation domain namespace.
+     *
+     * @return string
+     */
+    public function getTranslationDomain();
+
+    /**
+     * Returns the message without translation and with the placeholders replaced by the variables.
```

Comment:
I'm a bit confused by this, why "without translation"?
I also - just going by the names of the functions - don't know how I would render the message itself. I assume it'd be

```
$res = $foo->getValidationResult(); // pseudo

foreach ($res->getMessages() as $msg) {
    echo sprintf("<li>%s</li>", (string) $msg);
}
```

Alternatively to `(string) $msg` I'd like to have a clear getter that will give me the error message. Something like `$msg->getMessage()` or `$msg->getText()` or `$msg->getTranslatedMessage()`.

All we have in the interface for this is `getMessageTemplate()` (as far as I can understand it) which in my opinion isn't that clear.